Mergin/test/test_utils.py

```diff
@@ -90,6 +90,83 @@ def test_datum_shift_grids(self):
         self.assertEqual(len(grids), 1)
         self.assertTrue("uk_os_OSTN15_NTv2_OSGBtoETRS.tif" in grids or "OSTN15_NTv2_OSGBtoETRS.gsb" in grids)
 
+    def test_name_validation(self):
+        test_cases = [
+            ("project", True),
+            ("ProJect", True),
+            ("Pro123ject", True),
+            ("123PROJECT", True),
+            ("PROJECT", True),
+            ("project ", True),
+            ("pro ject", True),
+            ("proj-ect", True),
+            ("-project", True),
+            ("proj_ect", True),
+            ("proj.ect", True),
+            ("proj!ect", True),
+            (" project", False),
+            (".project", False),
+            ("proj~ect", False),
+            ("pro\ject", False),
+            ("pro/ject", False),
+            ("pro|ject", False),
+            ("pro+ject", False),
+            ("pro=ject", False),
+            ("pro>ject", False),
+            ("pro<ject", False),
+            ("pro@ject", False),
+            ("pro#ject", False),
+            ("pro$ject", False),
+            ("pro%ject", False),
+            ("pro^ject", False),
+            ("pro&ject", False),
+            ("pro*ject", False),
+            ("pro?ject", False),
+            ("pro:ject", False),
+            ("pro;ject", False),
+            ("pro,ject", False),
+            ("pro`ject", False),
+            ("pro'ject", False),
+            ('pro"ject', False),
+            ("projectz", True),
+            ("projectZ", True),
+            ("project0", True),
+            ("pro(ject", False),
+            ("pro)ject", False),
+            ("pro{ject", False),
+            ("pro}ject", False),
+            ("pro[ject", False),
+            ("pro]ject", False),
+            ("pro]ject", False),
+            ("CON", False),
+            ("NUL", False),
+            ("NULL", True),
+            ("PRN", False),
+            ("LPT0", False),
+            ("lpt0", True),
+            ("LPT1", False),
+            ("lpt1", True),
+            ("COM1", False),
+            ("com1", True),
+            ("AUX", False),
+            ("AuX", True),
+            ("projAUXect", True),
+            ("CONproject", True),
+            ("projectCON", True),
+            ("support", False),
+            ("helpdesk", False),
+            ("input", False),
+            ("lutraconsulting", False),
+            ("lutra", False),
+            ("merginmaps", False),
+            ("mergin", False),
+            ("admin", False),
+            ("sales", False),
```

Comment:
Can you please add the same Chinese/Hebrew project names as you did for mobile https://github.com/MerginMaps/input/pull/2449/files#diff-33c6f9baefe15bf8b0b47c49256319fad26445333a6bf988a5f797ef60dae96eR232-R233 ? thanks!